app/src/main/java/org/mozilla/focus/browser/integration/FullScreenIntegration.kt

```diff
@@ -0,0 +1,96 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.browser.integration
+
+import android.app.Activity
+import android.os.Build
+import android.view.View
+import android.view.WindowManager
+import androidx.annotation.RequiresApi
+import mozilla.components.browser.session.SessionManager
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.feature.session.FullScreenFeature
+import mozilla.components.feature.session.SessionUseCases
+import mozilla.components.support.base.feature.LifecycleAwareFeature
+import mozilla.components.support.base.feature.UserInteractionHandler
+import org.mozilla.focus.browser.DisplayToolbar
+
+class FullScreenIntegration(
+    val activity: Activity,
+    val store: BrowserStore,
+    val sessionManager: SessionManager,
+    tabId: String?,
+    private val toolbarView: DisplayToolbar,
+    private val statusBar: View
+) : LifecycleAwareFeature, UserInteractionHandler {
+    private val feature = FullScreenFeature(
+        store,
+        SessionUseCases(store, sessionManager),
```

Comment:
eventually we should consider moving them somewhere to the `Components` singleton too. But that can happen later, I guess.